snappy_pipeline/workflows/variant_calling/__init__.py

```diff
@@ -1022,6 +1025,76 @@ def get_resource_usage(self, action):
         )
 
 
+class BafFileGenerationStepPart(BaseStepPart):
+    """Base class for computing B allele fraction files.
+
+    One file is generated per sample in the output VCF files.
+    """
+
+    #: Step name
+    name = "baf_file_generation"
+
+    #: Class available actions
+    actions = ("run",)
+
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.base_path_out = (
+            "work/{mapper}.{var_caller}.{index_ngs_library}/report/baf/"
+            r"{mapper}.{var_caller}.{index_ngs_library}.{donor_ngs_library,[^\.]+}.baf"
+        )
+        # Build shortcut from index library name to pedigree
+        self.index_ngs_library_to_pedigree = OrderedDict()
+        for sheet in self.parent.shortcut_sheets:
+            self.index_ngs_library_to_pedigree.update(sheet.index_ngs_library_to_pedigree)
+
+    @dictify
+    def get_input_files(self, action):
+        """Return path to input files"""
+        # Validate action
+        self._validate_action(action)
+        # Return path to input VCF file
+        yield "vcf", (
+            "work/{mapper}.{var_caller}.{index_ngs_library}/out/"
+            "{mapper}.{var_caller}.{index_ngs_library}.vcf.gz"
+        )
+
+    @dictify
+    def get_output_files(self, action):
+        """Return output files that all germline variant calling sub steps must return (VCF +
+        TBI file)
+        """
+        # Validate action
+        self._validate_action(action)
+        ext_names = {"bw": ".bw", "bw_md5": ".bw.md5"}
+        for key, ext in ext_names.items():
+            yield key, self.base_path_out + ext
+
+    def get_log_file(self, action):
```

Comment:
Perhaps also log conda information? (requires changes to wrapper as well).

Reply:
done